app/commands/exercise/representation/create_or_update.rb

```diff
@@ -3,23 +3,64 @@ class Representation
     class CreateOrUpdate
       include Mandate
 
-      initialize_with :submission, :ast, :ast_digest, :mapping, :last_submitted_at
+      initialize_with :submission, :ast, :ast_digest, :mapping, :representer_version, :exercise_version, :last_submitted_at
 
       def call
-        representation = Exercise::Representation.find_create_or_find_by!(exercise:, ast_digest:) do |rep|
+        # First cache the old representation
+        @old_representation = Exercise::Representation.where(source_submission: submission).order('id DESC').first
+
+        @representation = Exercise::Representation.find_create_or_find_by!(
+          exercise:, ast_digest:, representer_version:, exercise_version:
+        ) do |rep|
           rep.source_submission = submission
           rep.ast = ast
           rep.mapping = mapping
           rep.last_submitted_at = last_submitted_at
         end
 
-        representation.update!(last_submitted_at:)
-        Exercise::Representation::UpdateNumSubmissions.defer(representation)
+        # Now copy the old feedback over if appropriate
+        update_feedback!
+        update_cache_columns!
 
         representation
       end
 
+      private
+      attr_reader :representation, :old_representation
+
       delegate :exercise, to: :submission
+
+      def update_feedback!
+        return unless old_representation&.has_feedback?
```

Comment:
Should we also guard against `old_representation == representation`? It looks like `exercise:, ast_digest:, representer_version:, exercise_version:` don't have to be different, right?

It might be that this additional check is not needed as it will be a no-op anyway, but it might be somewhat clearer to short circuit execution.